File: backend/app/schemas/admin_activity.py
"""Admin activity center schemas (ADMIN-NOTIFICATIONS-01B)."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

from app.core.admin_activity_constants import (
    ACTIVITY_FEED_LIMIT_DEFAULT,
    ACTIVITY_FEED_LIMIT_MAX,
    AdminActivityAlertSeverity,
    AdminActivityFeedCategory,
    AdminActivityFeedSeverity,
    AdminActivityHealthStatus,
)

AdminActivityCheckStatus = Literal["ok", "error", "unknown"]


class AdminActivityHealth(BaseModel):
    status: AdminActivityHealthStatus
    database: AdminActivityCheckStatus
    redis: AdminActivityCheckStatus


class AdminActivityAttentionSummary(BaseModel):
    critical: int = Field(ge=0)
    warning: int = Field(ge=0)
    total: int = Field(ge=0)
    healthy: bool


class AdminActivityAlert(BaseModel):
    id: str
    label: str
    description: str
    count: int = Field(ge=0)
    severity: AdminActivityAlertSeverity
    href: str
    category: Literal["moderation", "partners", "system"]


class AdminActivitySectionSummary(BaseModel):
    label: str
    count: int = Field(ge=0)
    severity: AdminActivityAlertSeverity


class AdminActivitySections(BaseModel):
    moderation: AdminActivitySectionSummary
    partners: AdminActivitySectionSummary
    system: AdminActivitySectionSummary


class AdminActivitySummaryResponse(BaseModel):
    generated_at: datetime
    read_only: bool = True
    health: AdminActivityHealth
    attention: AdminActivityAttentionSummary
    alerts: list[AdminActivityAlert]
    sections: AdminActivitySections


class AdminActivityFeedItem(BaseModel):
    id: str
    category: AdminActivityFeedCategory
    action: str
    title: str
    description: str
    actor_label: str
    target_label: str
    target_id: str
    href: str
    severity: AdminActivityFeedSeverity
    created_at: datetime


class AdminActivityFeedResponse(BaseModel):
    generated_at: datetime
    items: list[AdminActivityFeedItem]
    next_cursor: str | None = None


class AdminActivityFeedParams(BaseModel):
    limit: int = Field(default=ACTIVITY_FEED_LIMIT_DEFAULT, ge=1, le=ACTIVITY_FEED_LIMIT_MAX)
    cursor: str | None = None
    category: str = "all"

File: frontend/packages/types/src/admin-activity.ts
export type AdminActivityAlertSeverity = "healthy" | "warning" | "critical";
export type AdminActivityHealthStatus = "healthy" | "degraded" | "critical";
export type AdminActivityFeedSeverity = "info" | "success" | "warning" | "critical";
export type AdminActivityCheckStatus = "ok" | "error" | "unknown";

export type AdminActivityFeedCategory =
  | "partner"
  | "passport"
  | "offer"
  | "event"
  | "creator"
  | "moderation"
  | "staff"
  | "system"
  | "report";

export type AdminActivityFilterCategory =
  | "all"
  | AdminActivityFeedCategory;

export interface AdminActivityHealth {
  status: AdminActivityHealthStatus;
  database: AdminActivityCheckStatus;
  redis: AdminActivityCheckStatus;
}

export interface AdminActivityAttentionSummary {
  critical: number;
  warning: number;
  total: number;
  healthy: boolean;
}

export interface AdminActivityAlert {
  id: string;
  label: string;
  description: string;
  count: number;
  severity: AdminActivityAlertSeverity;
  href: string;
  category: "moderation" | "partners" | "system";
}

export interface AdminActivitySectionSummary {
  label: string;
  count: number;
  severity: AdminActivityAlertSeverity;
}

export interface AdminActivitySections {
  moderation: AdminActivitySectionSummary;
  partners: AdminActivitySectionSummary;
  system: AdminActivitySectionSummary;
}

export interface AdminActivitySummary {
  generated_at: string;
  read_only: boolean;
  health: AdminActivityHealth;
  attention: AdminActivityAttentionSummary;
  alerts: AdminActivityAlert[];
  sections: AdminActivitySections;
}

export interface AdminActivityFeedItem {
  id: string;
  category: AdminActivityFeedCategory;
  action: string;
  title: string;
  description: string;
  actor_label: string;
  target_label: string;
  target_id: string;
  href: string;
  severity: AdminActivityFeedSeverity;
  created_at: string;
}

export interface AdminActivityFeed {
  generated_at: string;
  items: AdminActivityFeedItem[];
  next_cursor: string | null;
}

export interface AdminActivityFeedParams {
  limit?: number;
  cursor?: string | null;
  category?: AdminActivityFilterCategory;
}
